pennylane/kernels/utils.py

```diff
@@ -120,3 +121,95 @@ def circuit(x1, x2):
             matrix[M * i + j] = kernel(X1[i], X2[j])
 
     return np.array(matrix).reshape((N, M))
+
+
+def kernel_eigensystem(X, kernel):
```

Comment:
`kernel_eigensystem()` and `task_weights()` work great for me. As you mentioned in the description, `task_model_alignment()`  is causing trouble as soon as there are `nan`s in the `task_weights` (resulting from the matrix diagonalization in `task_weights()`)